avocado/core/safeloader.py

```diff
@@ -30,38 +31,57 @@ class AvocadoModule(object):
     """
     Representation of a module that might contain avocado.Test tests
     """
-    __slots__ = 'path', 'test_import', 'mod_import', 'mod'
+    __slots__ = 'path', 'test_import', 'mod_import', 'mod', 'imported_objects'
 
     def __init__(self, path, test_import=False, mod_import=False):
         self.path = path
         self.test_import = test_import
         self.mod_import = mod_import
+        # Dict where:
+        #   key => object name how is visible from this module
+        #   value => Something-like a directory path to the import.
+        #            Basicall it's $path/$module/$variable, but depending
+        #            on import could be also $path/$module.
+        self.imported_objects = {}
         if os.path.isdir(path):
-            self.path = os.path.join(path, "__init__.py")
-        else:
-            self.path = path
-        with open(self.path) as source_file:
-            self.mod = ast.parse(source_file.read(), self.path)
+            path = os.path.join(path, "__init__.py")
```

Comment:
This simplification of the path assignment can be extract into its own commit.

Reply:
It could, similarly to number of other little refactors or improvements but do you think it's that important? I mean do you expect partial reverts or would it speedup the review process? I don't think so, anyway if you do I can extract it. It's a matter of a few clicks, commit message, another ci check and another review...

Reply:
What I meant is best shown here: 
https://github.com/avocado-framework/avocado/pull/3008/commits/485c008ee2a390d715bea1c6cbdd6c8995774fd0